src/Database/Schema/MysqlSchema.php

```diff
@@ -92,7 +92,7 @@ protected function _convertColumn($column)
             $precision = (int)$precision;
         }
 
-        if (in_array($col, ['date', 'time', 'datetime', 'timestamp'])) {
+        if (in_array($col, ['date', 'time', 'datetime', 'timestamp', 'year'])) {
             return ['type' => $col, 'length' => null];
```

Comment:
But the 'abstract' type will be 'year', will that convert back into the correct `CREATE TABLE` statement?